Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-current-contest.tsx

```diff
@@ -82,16 +140,57 @@ const CurrentContestsProvider = ({ children }: ICurrentContestsProviderProps) =>
     }, [ startContestData ]);
 
     useEffect(() => {
-        if (isNil(contestToStart)) {
+        if (isNil(registerForContestParams)) {
             return;
         }
 
         (async () => {
             startLoading();
-            await startContest();
+            await registerForContest();
             stopLoading();
         })();
-    }, [ contestToStart, startContest, startLoading, stopLoading ]);
+    }, [ registerForContest, registerForContestParams, startLoading, stopLoading ]);
+
+    useEffect(() => {
+        if (isNil(registerForContestData)) {
+            return;
+        }
+
+        const responseData = registerForContestData as IRegisterForContestResponseType;
+        const { requirePassword: responseRequirePassword } = responseData;
+
+        setContest({ id: responseData.id, name: responseData.name } as IContestType);
+        setRequirePassword(responseRequirePassword);
+    }, [ registerForContestData ]);
```

Comment:
As I understand, the concept behind useMemo is to optimize expensive CPU calculations instead of recalculating again. In this case, it seems to be macro optimization, but on the other hand, it will preserve executing the same logic multiple times.
Both approaches seem fine to me.
@shsimeonova @Minkov can't advise on that. Above are my thoughts.

Reply:
All above is correct.
My proposal is based not on performance optimizations, but on semantics. `useEffect` implies logic, while `useMemo` implies a variable. This is variable, so...

Reply:
I don't see a big change in semantics. I think we generally have a problem with the boiler code assosiated with parameters.

Reply:
Boiler plate code assosiated with parameters?